webhook/webhook.go

```diff
@@ -40,37 +40,57 @@ func StartWebhook(ctx context.Context, webhookType string, clients *client.Clien
 	}()
 
 	logrus.Infof("Waiting for %v webhook to become ready", webhookType)
+	if !isServiceAvailable(webhookLocalEndpoint) {
+		return fmt.Errorf("%v webhook is not ready on localhost after %v sec", webhookType, defaultStartTimeout)
+	}
+	logrus.Infof("Started longhorn %s webhook server on localhost", webhookType)
+	return nil
+}
+
+func CheckWebhookServiceAvailability(webhookType string) error {
+	var webhookServiceEndpoint string
+	switch webhookType {
+	case types.WebhookTypeAdmission:
+		webhookServiceEndpoint = fmt.Sprintf("https://longhorn-admission-webhook.longhorn-system.svc:%d/v1/healthz", types.DefaultAdmissionWebhookPort)
```

Comment:
Don't hard code the namespace here, even though the default is `longhorn-system` and we don't commit other namespaces will be supported.

You can use the follow function to get the namespace.

[/util/util.go#L581-L589](https://github.com/longhorn/longhorn-manager/blob/1956c63e2c5d55f03c2afba8e2bbf44a61218e59/util/util.go#L581-L589)
```go
func GetNamespace(key string) string {
	namespace := os.Getenv(key)
	if namespace == "" {
		logrus.Warnf("Failed to detect pod namespace, environment variable %v is missing, "+
			"using default namespace", key)
		namespace = corev1.NamespaceDefault
	}
	return namespace
}
```

Besides, use the existing constants (like `AdmissionWebhookServiceName`) instead of raw string here. Please review this in other places as well.

Reply:
Fixed, thanks!